stablehlo/reference/InterpreterScope.h

```diff
@@ -0,0 +1,64 @@
+/* Copyright 2022 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_REFERENCE_SCOPE_H_
+#define STABLEHLO_REFERENCE_SCOPE_H_
+
+#include "stablehlo/reference/Tensor.h"
+
+namespace mlir {
+namespace stablehlo {
+
+/// Represents the runtime scope corresponding to a region of a program under
+/// evaluation. Holds (1) mapping from SSA values, defined in the current
+/// region, to their evaluated runtime `Tensor` values, and (2) handle to
+/// `InterpreterScope` object corresponding to the syntactically enclosing
+/// region.
+///
+/// Note:
+/// 1. A `InterpreterScope` object is instantiated every time a region is
+///    evaluated.
+/// 2. A `InterpreterScope` object treats the `parentScope` as immutable to
+///    align with the fact that a StableHLO program, in pure SSA form (without
+///    memory allocation/load/store ops), disallows mutating the `parentScope`
+///    object from within a region.
+class InterpreterScope {
+ public:
+  InterpreterScope(const InterpreterScope *const pScope)
```

Comment:
Let's call this parameter `parent`? Both [the LLVM style guide](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly) and [the Google style guide](https://google.github.io/styleguide/cppguide.html) discourage the use of abbreviations unless they are well known. I would say that abbreviating "parent" to "p" isn't particularly well known. Same for the constructor below.

Reply:
I am calling this `parentScope` just to differentiate it with a member variable with the same name. Just for clarity.

Reply:
The current convention that we're using for this follows Google style: call parameters `foo` and corresponding fields `foo_`.